src/EventTickets/ViewModels/EventDetails.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace Give\EventTickets\ViewModels;
+
+use Give\DonationForms\Models\DonationForm;
+use Give\EventTickets\DataTransferObjects\EventTicketTypeData;
+use Give\EventTickets\Models\Event;
+use Give\EventTickets\Models\EventTicketType;
+
+/**
+ * @unreleased
+ */
+class EventDetails
+{
+
+    /**
+     * @unreleased
+     * @var Event
+     */
+    protected $event;
+
+    /**
+     * @unreleased
+     */
+    public function __construct(Event $event)
+    {
+        $this->event = $event;
+    }
+
+    /**
+     * @unreleased
+     */
+    public function exports(): array
+    {
+        return array_merge($this->event->toArray(), [
+
+            'ticketTypes' => array_map(function (EventTicketType $ticketType) {
+                return EventTicketTypeData::make($ticketType)->toArray();
+            }, $this->event->ticketTypes),
```

Comment:
Shouldn't this be calling the method instead?

Reply:
Calling `->ticketType` uses a caching layer added by model relationships. The relationship is queried and cached for later calls.

Calling `->ticketType()` as a method returns a `ModelQueryBuilder`, which is called under the hood when getting the relationship as a property.